sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -245,16 +245,58 @@ case class GpuDivide(left: Expression, right: Expression) extends GpuDivModLike
 
   override def symbol: String = "/"
 
-  override def binaryOp: BinaryOp = BinaryOp.TRUE_DIV
+  override def binaryOp: BinaryOp = (left.dataType, right.dataType) match {
+    case (_: DecimalType, _: DecimalType) =>  BinaryOp.DIV
+    case _ => BinaryOp.TRUE_DIV
+  }
 
   override def outputTypeOverride: DType =
     GpuColumnVector.getNonNestedRapidsType(dataType)
 
   // Override the output type as a special case for decimal
   override def dataType: DataType = (left.dataType, right.dataType) match {
-    case (l: DecimalType, r: DecimalType) =>  GpuDivideUtil.decimalDataType(l, r)
+    case (l: DecimalType, r: DecimalType) => GpuDivideUtil.decimalDataType(l, r)
     case _ => super.dataType
   }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    (left.dataType, right.dataType) match {
+      case (_: DecimalType, r: DecimalType) => {
+        val outType = dataType.asInstanceOf[DecimalType]
+        val newType = DecimalType(outType.precision, outType.scale + r.scale)
```

Comment:
Can you please add a comment to explain the rules here about what is needed for the divide to work? It looks like from the CUDF code that for a divide the output scale must be `lhs.scale - rhs.scale = outputScale`.  I did a much of math and I think casting the lhs up is going to be fine until we hit 128-bit support for decimal and need to potentially check for overflow, so it would be good to have a comment about that here too.

Reply:
OK, I will do that. I thought it was unnecessary as we are already checking for it in `tagExprForGpu`

Reply:
nevermind, we are only checking for the intermediary result. 

I think we should add a check for `GpuDivideUtil.decimalDataType + r.scale` instead of what we are doing now? What do you think?